#!/usr/bin/env python
PACKAGE = "ubiquity_motor"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("PID_P",    int_t,    0B00000001, "Param P", 9000,  0, 100000)
gen.add("PID_I",    int_t,    0B00000010, "Param I", 14,  0, 10000)
gen.add("PID_D",    int_t,    0B00000100, "Param D", -60,  -100000, 100000)
gen.add("PID_C",    int_t,    0B00001000, "Denominator", 1000,  0, 100000)
gen.add("PID_W",    int_t,    0B00010000, "D Window", 100,  0, 200)
gen.add("PID_V",    int_t,    0B00100000, "Param V", 0, -10000000, 10000000)
gen.add("MAX_PWM",  int_t,    0B01000000, "Max PWM", 230, 100, 5000)

exit(gen.generate(PACKAGE, "motor_node", "PID"))
